src/interfaces/highs_c_api.cpp

```diff
@@ -1198,6 +1198,16 @@ void Highs_resetGlobalScheduler(HighsInt blocking) {
   Highs::resetGlobalScheduler(blocking);
 }
 
+
+void* Highs_mallocOrExit(unsigned long long size, const char* var_name) {
+    void* ptr = malloc(size);
+    if (ptr == NULL) {
+        fprintf(stderr, "Out of memory when allocating %s\n", var_name);
+        exit(EXIT_FAILURE);
```

Comment:
I'm no developer of HiGHS, but please do not add calls to exit() that might be reached when HiGHS is used as library - there is no way to handle the out of memory error in that case.

Example: HiGHS is used as library within a web server. With this code, the web server will simply stop if one request triggers an out of memory error...

Reply:
Wouldn't it crash if `malloc` failed anyway? This just makes it easier to debug instead of failing at a later stage due to a nullptr. Also although it's part of the C API, users can still make calls to `malloc`if they wish to handle memory differently.

Reply:
These `malloc` calls will only fail if there's insufficient memory available. Having HiGHS fail cleanly in such circumstances is on the ToDo list, but it is very unusual so we've not prioritised it.

As you see @HaoZeke (and also with reference to #1368), having HiGHS handle errors via exceptions and exit will meet with valid objections.